test/emailpassword/signinFeature.test.js

```diff
@@ -39,3 +56,1315 @@
+                    ],
+                })
+                .expect(400)
+                .end((err, res) => {
+                    if (err) {
+                        resolve(undefined);
+                    } else {
+                    }
+                    resolve(JSON.parse(res.text));
+                })
+        );
+        assert(response.message === "Are you sending too many / too few formFields?");
+    });
+
+    //Provide invalid (wrong syntax) email and wrong password, and you should get form field error
+
+    /*
+    passing email with valid syntax and correct password will cause the test to fail
+    */
+    it("test invalid email and wrong password", async function () {
+        await startST();
+
+        STExpress.init({
+            supertokens: {
+                connectionURI: "http://localhost:8080",
+            },
+            appInfo: {
+                apiDomain: "api.supertokens.io",
+                appName: "SuperTokens",
+                websiteDomain: "supertokens.io",
+            },
+            recipeList: [EmailPassword.init(), Session.init()],
+        });
+        const app = express();
+
+        app.use(STExpress.middleware());
+
+        app.use(STExpress.errorHandler());
+
+        await new Promise((resolve) =>
+            request(app)
+                .post("/auth/signup")
+                .send({
+                    formFields: [
+                        {
+                            id: "password",
+                            value: "validpass123",
+                        },
+                        {
+                            id: "email",
+                            value: "random@gmail.com",
+                        },
+                    ],
+                })
+                .expect(200)
+                .end((err, res) => {
+                    if (err) {
+                        resolve(undefined);
+                    } else {
+                        resolve(res);
+                    }
+                })
+        );
+
+        let response = await new Promise((resolve) =>
+            request(app)
+                .post("/auth/signin")
+                .send({
+                    formFields: [
+                        {
+                            id: "password",
+                            value: "invalid",
+                        },
+                        {
+                            id: "email",
+                            value: "randomgmail",
+                        },
+                    ],
+                })
+                .expect(200)
+                .end((err, res) => {
+                    if (err) {
+                        resolve(undefined);
+                    } else {
+                    }
+                    resolve(JSON.parse(res.text));
+                })
+        );
+        assert(response.status === "FIELD_ERROR");
+        assert(response.formFields[0].error === "Email is invalid");
+        assert(response.formFields[0].id === "email");
```

Comment:
it should also have another element! formFields[1] should contain an error for the password! That's the point of this test.

Reply:
Im testing the signInFeature, the signInFeature throws "FIELD_ERROR" for email syntax error not for password

Reply:
Okay. So then do this with signUp and check that it returns both the errors, and with sign in, the size of `formFields` is 1

Reply:
@jscyo you missed this comment

Reply:
ive added the check that formFields should be 1, ill be doing the signUp check in the signUpFeature test